### PR TITLE
Separation of UI and Logic in Component `Slider`

### LINKED ISSUE #3288: Separation of UI and Logic in the `Slider` Component
### Description

Separate logic into separate files such as `use-xxx.ts`.
In addition, modify the hooks to function on their own.
The following components need to be addressed.

- `Slider`
- `RangeSlider`
- `HueSlider`
- AlphaSlider`
- SaturationSlider`

### Additional Information

Create a new branch from `v2.0` branch for development.
https://react-spectrum.adobe.com/react-aria/useSlider.html#useslider

### PR DESCRIPTION
**Pull request by EGASHIRAAkihide**
Closes #3288 

## Description

Separate UI and Logic in `Slider.tsx` and move Logic to `use-slider.tsx`

## Current behavior (updates)

No Change

## New behavior

No Change

## Is this a breaking change (Yes/No):

No

## Additional Information

このPRのmerge先は `v2.0` に設定してあります。
The merge target for this PR is set to v2.0

https://github.com/yamada-ui/yamada-ui/issues/3288#issuecomment-2446900464

### CODE REVIEW

**Comment by EGASHIRAAkihide** [comment]
### メモ：interface `SliderOptions` の配置に関して

このPRを作成するにあたり、他コンポーネントの実装を参考に分割いたしました。
`XXOptions` との名称で作成されているinterfaceのうち、`ripple`のみ、`use-ripple.tsx`に配置されておりました。
それ以外のcomponentに関しましては、基本的にUIファイルの方に型定義されているため、`ripple`のみロジックファイル側に`XXOptions` が格納されているか、理由が知りたいです。これは意図的に配置されているからでしょうか？

参考：`/packages/components/ripple/src/use-ripple.tsx`

In creating this PR, I referenced the implementation of other components and separated them accordingly. Among the interfaces created with the name `XXOptions`, only ripple is located in `use-ripple.tsx`. As for other components, their type definitions are generally in the UI files, so I would like to understand why only ripple has `XXOptions` stored on the logic file side. Was this placement intentional?

**Review by kaehehehe — COMMENTED** [review]
Thank you for the PR. @EGASHIRAAkihide 

Currently, we are planning to migrate the logic related to state and props to hooks in all components.

File: .changeset/chilly-ghosts-rhyme.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@yamada-ui/slider": major
```
Comment (kaehehehe) [inline .changeset/chilly-ghosts-rhyme.md:2]: Please change major to patch.

**Review by kaehehehe — CHANGES_REQUESTED** [review]

File: packages/components/slider/src/use-slider.tsx
Comment (kaehehehe) [inline packages/components/slider/src/use-slider.tsx:1]: I'm sorry, but could you change the extension of the hook from `.tsx` to `.tx`?

**Review by kaehehehe — APPROVED** [review]
Thanks for contributing!

File: .changeset/witty-rocks-search.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@yamada-ui/slider": patch
```
Comment (kaehehehe) [inline .changeset/witty-rocks-search.md:2]: Sorry! 😢 The changeset is not needed for this PR, as it is related to refactoring. Please delete this file. @EGASHIRAAkihide

**Review by kaehehehe — CHANGES_REQUESTED** [review]

**Comment by hirotomoyamada** [comment]
@EGASHIRAAkihide 

If this PR is incorporated first, conflicts will occur in other work, so close this PR.

Please submit this PR again after the other work is completed.

**Pull request closed without merging**